### PR TITLE
updates to CI java http server tutorial

### PR DESCRIPTION
**Pull request by jimsheldon**
more yaml-based

use include for docker hub connector

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by ravilach — APPROVED** [review]
Thanks for adding to shared fold and trying out the component capability. Looks good to me.

**Comment by ravilach** [comment]
Was able to merge in 181.

File: tutorials/build-code/ci-java-http-server.md [hunk 1]
```diff
@@ -35,136 +39,84 @@ import TabItem from '@theme/TabItem';
...
-1. Select **Add Step**. In the **Step Library** dialog, select **Background** from the list.
-2. In the **Name** field, enter `Run Java HTTP Server`.
-3. Expand the **Additional Configuration** section, select the **Container Registry** field, and then select your Docker Hub connector.
-4. In the **Image** field, etner `your_user/jhttp:<+pipeline.sequenceId>`. Replace `your_user` with your Docker Hub username.
-5. Under **Port Bindings**, select **+ Add**, and then enter `8888` in the **Host Port** and **Container Port** fields.
-
-   :::info
-
-   This exposes port `8888` from the running container to the host operating system. This allows the connection test step to reach the application at `localhost:8888`.
+### Modify pipeline
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:67]:
```suggestion
### Modify the pipeline
```

File: tutorials/build-code/ci-java-http-server.md [hunk 2]
```diff
@@ -35,136 +39,84 @@ import TabItem from '@theme/TabItem';
...
 
-   :::
+Select **Pipelines** from the menu on the left, then select your **jhttp** pipeline from the list.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:69]:
```suggestion
From the left pane, select **Pipelines**, and then select your **jhttp** pipeline from the list.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 3]
```diff
@@ -243,20 +195,22 @@ pipeline:
...
-        connectorRef: account.Github_OAuth_XYZ123
-        repoName: your_user/jhttp
-        build: <+input>
 ```
 
-</p>
-</details>
+:::info
+
+This configuration expects the Docker Hub connector ID to be `Docker_Hub`. If your connector ID is different, replace `Docker_Hub` with the correct ID.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:202]:
```suggestion
This configuration requires the Docker Hub connector ID to be `Docker_Hub`. If your connector ID is different, replace `Docker_Hub` with the correct ID.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 4]
```diff
@@ -243,20 +195,22 @@ pipeline:
...
+
+:::
+
+Click the **Save** button in the YAML editor.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:206]:
```suggestion
Select **Save** in the YAML editor.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 5]
```diff
@@ -243,20 +195,22 @@ pipeline:
...
 
 ## Run your pipeline
 
 1. In the **Pipeline Studio**, select **Run**.
+2. Enter your Docker Hub username in the `DOCKERHUB_USERNAME` field.
 2. Select **Git Branch** as the **Build Type**, and then enter `main` in the **Branch Name** field.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:212]:
```suggestion
2. In the **Build Type** field, select **Git Branch**, and then enter **main** in the **Branch Name** field.
```

File: tutorials/build-code/ci-java-http-server.md [hunk 6]
```diff
@@ -243,20 +195,22 @@ pipeline:
...
 3. Select **Run Pipeline**.
-4. Observe each step of the pipeline execution. When the first stage completes, test results appear in the **Tests** tab. When the second stage completes, you should see the successful `curl` command in the final step.
+4. Observe each step of the pipeline execution. When the first stage completes, test results appear in the **Tests** tab.
```
Comment (KimberlyFields) [inline tutorials/build-code/ci-java-http-server.md:214]:
```suggestion
4. Observe each step of the pipeline execution. When the first stage completes, test results appear on the **Tests** tab.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+1. Expand the **Project Setup** menu on the left, then select **Connectors**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:1]:
```suggestion
1. From the left pane, expand the **Project Setup**, and then select **Connectors**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+2. Click **+ New Connector**, and then select **Docker Registry**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:2]:
```suggestion
2. Select **+ New Connector**, and then select **Docker Registry**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+3. In the **Name** field, enter `Docker Hub`, and then select **Continue**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:3]:
```suggestion
3. In the **Name** field, enter **Docker Hub**, and then select **Continue**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+4. In the **Docker Registry URL** field, enter `https://index.docker.io/v2/`.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:4]:
```suggestion
4. In the **Docker Registry URL** field, enter **https://index.docker.io/v2/**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+5. For **Provider Type**, select **DockerHub**.
+6. In the **Username** field, enter your Docker Hub username.
+7. In the **Password** field, select **Create or Select a Secret**. Follow the prompts to add your Docker Hub access token, then click **Continue**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:7]:
```suggestion
7. In the **Password** field, select **Create or Select a Secret**. Follow the prompts to add your Docker Hub access token, and then click **Continue**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+8. At the next screen, select **Connect through Harness Platform**, then select **Save and Continue**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:8]:
```suggestion
8. At the next screen, select **Connect through Harness Platform**, and then select **Save and Continue**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+9. Harness will perform a validation test, then click **Finish**.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:9]:
```suggestion
9. After Harness performs a validation test, select **Finish**.
```

File: tutorials/shared/dockerhub-connector-includes.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+10. In your list of connectors, note the ID of the connector that you just created. If you named the connecotr `Docker Hub`, the ID should be `Docker_Hub`.
```
Comment (KimberlyFields) [inline tutorials/shared/dockerhub-connector-includes.md:10]:
```suggestion
10. In your list of connectors, note the ID of the connector that you just created. If you named the connector `Docker Hub`, the ID should be `Docker_Hub`.
```

**Review by KimberlyFields — CHANGES_REQUESTED** [review]

**Comment by jimsheldon** [comment]
Ping for @KimberlyFields if this is ready to merge 🙏🏻

**Comment by KimberlyFields** [comment]
LGTM. @ravilach you didn't approve the PR. Do you want to give it one more review before merging?

**Pull request merged**